File: frontend/src/stores/eventTypes.ts
/**
 * Unified Event Types - matches backend event_types.go
 * Core types for the new event system with SQLite persistence
 */

import React from 'react';
import {
  // Source icons
  FileTextOutlined,
  GlobalOutlined,
  MobileOutlined,
  AppstoreOutlined,
  PictureOutlined,
  AimOutlined,
  NodeIndexOutlined,
  BarChartOutlined,
  ToolOutlined,
  CheckCircleOutlined,
  // Level icons
  UnorderedListOutlined,
  BugOutlined,
  InfoCircleOutlined,
  WarningOutlined,
  CloseCircleOutlined,
  StopOutlined,
  // Event type icons
  PlayCircleOutlined,
  FlagOutlined,
  ApiOutlined,
  ThunderboltOutlined,
  WifiOutlined,
  DesktopOutlined,
  PauseCircleOutlined,
  FileOutlined,
  ExclamationCircleOutlined,
  ClockCircleOutlined,
  HighlightOutlined,
  ExpandOutlined,
  CaretRightOutlined,
  CheckOutlined,
  BorderOutlined,
  FieldTimeOutlined,
} from '@ant-design/icons';

// ========================================
// Event Source - 事件来源
// ========================================

export type EventSource =
  | 'logcat'     // 设备日志
  | 'network'    // 网络请求
  | 'device'     // 设备状态
  | 'app'        // 应用生命周期
  | 'ui'         // UI 状态变化
  | 'touch'      // 触摸事件
  | 'workflow'   // 自动化流程
  | 'perf'       // 性能指标
  | 'system'     // 系统事件
  | 'assertion'; // 断言结果

// ========================================
// Event Category - 事件大类
// ========================================

export type EventCategory =
  | 'log'
  | 'network'
  | 'state'
  | 'interaction'
  | 'automation'
  | 'diagnostic';

// ========================================
// Event Level - 事件级别
// ========================================

export type EventLevel =
  | 'verbose'
  | 'debug'
  | 'info'
  | 'warn'
  | 'error'
  | 'fatal';

// ========================================
// UnifiedEvent - 统一事件结构
// ========================================

export interface UnifiedEvent {
  // 标识字段
  id: string;
  sessionId: string;
  deviceId: string;

  // 时间字段
  timestamp: number;      // Unix 毫秒 (绝对时间)
  relativeTime: number;   // 相对 Session 开始的毫秒偏移
  duration?: number;      // 持续时间 (ms)

  // 分类字段
  source: EventSource;
  category: EventCategory;
  type: string;           // 具体类型 (如 "http_request", "activity_start")
  level: EventLevel;

  // 内容字段
  title: string;
  summary?: string;

  // 扩展数据 (JSON)
  data?: any;

  // Legacy alias for backward compatibility
  detail?: any;

  // 关联字段
  parentId?: string;
  stepId?: string;
  traceId?: string;

  // 聚合信息
  aggregateCount?: number;
  aggregateFirst?: number;
  aggregateLast?: number;
}

// ========================================
// Session Config - Session 启动配置
// ========================================

export interface LogcatConfig {
  enabled: boolean;
  packageName?: string;
  preFilter?: string;
  excludeFilter?: string;
}

export interface RecordingConfig {
  enabled: boolean;
  quality?: 'low' | 'medium' | 'high';
}

export interface ProxyConfig {
  enabled: boolean;
  port?: number;
  mitmEnabled?: boolean;
}

export interface MonitorConfig {
  enabled: boolean; // 设备状态监控 (电池、网络、屏幕、应用生命周期)
}

export interface SessionConfig {
  logcat: LogcatConfig;
  recording: RecordingConfig;
  proxy: ProxyConfig;
  monitor: MonitorConfig;
}

// 默认配置 - 全部启用
export const defaultSessionConfig: SessionConfig = {
  logcat: { enabled: true, packageName: '' },
  recording: { enabled: true, quality: 'medium' },
  proxy: { enabled: true, port: 8080, mitmEnabled: true },
  monitor: { enabled: true },
};

// ========================================
// DeviceSession - 设备会话
// ========================================

export interface DeviceSession {
  id: string;
  deviceId: string;
  type: string;       // "manual", "workflow", "recording", "debug", "auto"
  name: string;
  startTime: number;  // Unix ms
  endTime: number;    // 0 = active
  status: 'active' | 'completed' | 'failed' | 'cancelled';
  eventCount: number;

  // Session config (启用的功能)
  config?: SessionConfig;

  // Recording info
  videoPath?: string;
  videoDuration?: number;
  videoOffset?: number;

  // Metadata
  metadata?: Record<string, any>;
}

// ========================================
// Type-Specific Data Structures
// ========================================

// Logcat 事件数据
export interface LogcatData {
  tag: string;
  message: string;
  androidLevel: string; // V/D/I/W/E/F
  pid?: number;
  tid?: number;
  packageName?: string;
  raw?: string;
}

// ========================================
// Query Types
// ========================================

export interface EventQuery {
  sessionId?: string;
  deviceId?: string;
  sources?: EventSource[];
  categories?: EventCategory[];
  types?: string[];
  levels?: EventLevel[];
  startTime?: number;   // 相对时间 (ms)
  endTime?: number;
  searchText?: string;
  parentId?: string;
  stepId?: string;
  traceId?: string;
  limit?: number;
  offset?: number;
  orderDesc?: boolean;  // true = 时间倒序
  includeData?: boolean; // true = 加载完整 event_data（网络请求URL等）
}

export interface EventQueryResult {
  events: UnifiedEvent[];
  total: number;
  hasMore: boolean;
}

// ========================================
// Time Index & Bookmarks
// ========================================

export interface TimeIndexEntry {
  second: number;        // 相对 session 开始的秒数
  eventCount: number;
  firstEventId: string;
  hasError: boolean;
}

export interface Bookmark {
  id: string;
  sessionId: string;
  relativeTime: number;  // 相对 session 开始的毫秒
  label: string;
  color?: string;
  type: 'user' | 'error' | 'milestone' | 'assertion_fail';
  createdAt: number;
}

// ========================================
// UI Display Helpers
// ========================================

// Source 显示配置 - 使用 Ant Design 预设颜色名称以适配主题
export const sourceConfig: Record<EventSource, { label: string; icon: string; iconComponent: React.ReactNode; color: string }> = {
  logcat: { label: 'Logcat', icon: '📝', iconComponent: React.createElement(FileTextOutlined), color: 'green' },
  network: { label: 'Network', icon: '🌐', iconComponent: React.createElement(GlobalOutlined), color: 'purple' },
  device: { label: 'Device', icon: '📱', iconComponent: React.createElement(MobileOutlined), color: 'cyan' },
  app: { label: 'App', icon: '📦', iconComponent: React.createElement(AppstoreOutlined), color: 'blue' },
  ui: { label: 'UI', icon: '🖼️', iconComponent: React.createElement(PictureOutlined), color: 'magenta' },
  touch: { label: 'Touch', icon: '👆', iconComponent: React.createElement(AimOutlined), color: 'orange' },
  workflow: { label: 'Workflow', icon: '⚙️', iconComponent: React.createElement(NodeIndexOutlined), color: 'geekblue' },
  perf: { label: 'Perf', icon: '📊', iconComponent: React.createElement(BarChartOutlined), color: 'gold' },
  system: { label: 'System', icon: '🔧', iconComponent: React.createElement(ToolOutlined), color: 'default' },
  assertion: { label: 'Assert', icon: '✅', iconComponent: React.createElement(CheckCircleOutlined), color: 'green' },
};

// Category 显示配置 - 使用 Ant Design 预设颜色名称以适配主题
export const categoryConfig: Record<EventCategory, { label: string; color: string }> = {
  log: { label: 'Log', color: 'green' },
  network: { label: 'Network', color: 'purple' },
  state: { label: 'State', color: 'blue' },
  interaction: { label: 'Interaction', color: 'orange' },
  automation: { label: 'Automation', color: 'geekblue' },
  diagnostic: { label: 'Diagnostic', color: 'default' },
};

// Level 显示配置 - 使用 Ant Design 预设颜色名称以适配主题
export const levelConfig: Record<EventLevel, { label: string; color: string; icon: string; iconComponent: React.ReactNode; priority: number }> = {
  verbose: { label: 'Verbose', color: 'default', icon: '📋', iconComponent: React.createElement(UnorderedListOutlined), priority: 0 },
  debug: { label: 'Debug', color: 'default', icon: '🔧', iconComponent: React.createElement(BugOutlined), priority: 1 },
  info: { label: 'Info', color: 'blue', icon: 'ℹ️', iconComponent: React.createElement(InfoCircleOutlined), priority: 2 },
  warn: { label: 'Warn', color: 'gold', icon: '⚠️', iconComponent: React.createElement(WarningOutlined), priority: 3 },
  error: { label: 'Error', color: 'red', icon: '❌', iconComponent: React.createElement(CloseCircleOutlined), priority: 4 },
  fatal: { label: 'Fatal', color: 'red', icon: '💀', iconComponent: React.createElement(StopOutlined), priority: 5 },
};

// Event Type 显示配置
export const eventTypeConfig: Record<string, { label: string; icon: string; iconComponent: React.ReactNode }> = {
  // Session
  session_start: { label: 'Session Start', icon: '🚀', iconComponent: React.createElement(PlayCircleOutlined, { style: { color: '#52c41a' } }) },
  session_end: { label: 'Session End', icon: '🏁', iconComponent: React.createElement(FlagOutlined) },

  // Logcat
  logcat: { label: 'Log', icon: '📝', iconComponent: React.createElement(FileTextOutlined) },
  logcat_aggregated: { label: 'Logs', icon: '📝', iconComponent: React.createElement(FileTextOutlined) },

  // Network
  http_request: { label: 'HTTP Request', icon: '🌐', iconComponent: React.createElement(ApiOutlined) },
  websocket_message: { label: 'WebSocket', icon: '🔌', iconComponent: React.createElement(ApiOutlined, { style: { color: '#722ed1' } }) },

  // Device State
  battery_change: { label: 'Battery', icon: '🔋', iconComponent: React.createElement(ThunderboltOutlined, { style: { color: '#faad14' } }) },
  network_change: { label: 'Network', icon: '📶', iconComponent: React.createElement(WifiOutlined) },
  screen_change: { label: 'Screen', icon: '📱', iconComponent: React.createElement(DesktopOutlined) },

  // App Lifecycle
  app_start: { label: 'App Start', icon: '▶️', iconComponent: React.createElement(CaretRightOutlined, { style: { color: '#52c41a' } }) },
  app_stop: { label: 'App Stop', icon: '⏹️', iconComponent: React.createElement(PauseCircleOutlined) },
  activity_start: { label: 'Activity', icon: '📄', iconComponent: React.createElement(FileOutlined, { style: { color: '#1890ff' } }) },
  activity_stop: { label: 'Activity Stop', icon: '📄', iconComponent: React.createElement(FileOutlined) },
  app_crash: { label: 'Crash', icon: '💥', iconComponent: React.createElement(ExclamationCircleOutlined, { style: { color: '#ff4d4f' } }) },
  app_anr: { label: 'ANR', icon: '⏰', iconComponent: React.createElement(ClockCircleOutlined, { style: { color: '#fa8c16' } }) },

  // Touch
  touch: { label: 'Touch', icon: '👆', iconComponent: React.createElement(HighlightOutlined) },
  gesture: { label: 'Gesture', icon: '🤚', iconComponent: React.createElement(ExpandOutlined) },

  // Workflow
  workflow_start: { label: 'Workflow Start', icon: '▶️', iconComponent: React.createElement(CaretRightOutlined, { style: { color: '#2f54eb' } }) },
  workflow_step_start: { label: 'Step Start', icon: '🔷', iconComponent: React.createElement(BorderOutlined, { style: { color: '#1890ff' } }) },
  workflow_step_end: { label: 'Step End', icon: '✅', iconComponent: React.createElement(CheckOutlined, { style: { color: '#52c41a' } }) },
  workflow_complete: { label: 'Workflow Complete', icon: '🏁', iconComponent: React.createElement(FlagOutlined, { style: { color: '#52c41a' } }) },
  workflow_error: { label: 'Workflow Error', icon: '❌', iconComponent: React.createElement(CloseCircleOutlined, { style: { color: '#ff4d4f' } }) },

  // Performance
  perf_sample: { label: 'Perf Sample', icon: '📊', iconComponent: React.createElement(BarChartOutlined) },

  // Assertion
  assertion_result: { label: 'Assertion', icon: '✅', iconComponent: React.createElement(CheckCircleOutlined, { style: { color: '#52c41a' } }) },

  // Recording
  recording_start: { label: 'Recording Start', icon: '🔴', iconComponent: React.createElement(FieldTimeOutlined, { style: { color: '#ff4d4f' } }) },
  recording_end: { label: 'Recording End', icon: '⏹️', iconComponent: React.createElement(PauseCircleOutlined) },
};

// ========================================
// Utility Functions
// ========================================

/**
 * 格式化相对时间为可读字符串
 */
export function formatRelativeTime(ms: number): string {
  const totalSeconds = Math.floor(ms / 1000);
  const hours = Math.floor(totalSeconds / 3600);
  const minutes = Math.floor((totalSeconds % 3600) / 60);
  const seconds = totalSeconds % 60;
  const millis = ms % 1000;

  if (hours > 0) {
    return `${hours}:${pad(minutes)}:${pad(seconds)}.${pad3(millis)}`;
  }
  return `${minutes}:${pad(seconds)}.${pad3(millis)}`;
}

/**
 * 格式化绝对时间戳为时间字符串
 */
export function formatTimestamp(timestamp: number): string {
  const date = new Date(timestamp);
  return date.toLocaleTimeString('zh-CN', {
    hour: '2-digit',
    minute: '2-digit',
    second: '2-digit',
  }) + '.' + pad3(date.getMilliseconds());
}

/**
 * 格式化持续时间 (毫秒 → 紧凑格式: "381ms", "5.0s", "2m 30s")
 */
export function formatDuration(ms: number): string {
  if (ms < 1000) return `${ms}ms`;
  if (ms < 60000) return `${(ms / 1000).toFixed(1)}s`;
  const mins = Math.floor(ms / 60000);
  const secs = Math.floor((ms % 60000) / 1000);
  return `${mins}m ${secs}s`;
}

/**
 * 格式化持续时间 (毫秒 → 人类可读: "3s", "2m 3s", "1h 2m 3s")
 */
export function formatDurationHMS(ms: number): string {
  if (!ms || ms <= 0) return '-';
  const seconds = Math.floor(ms / 1000);
  const minutes = Math.floor(seconds / 60);
  const hours = Math.floor(minutes / 60);

  if (hours > 0) {
    return `${hours}h ${minutes % 60}m ${seconds % 60}s`;
  } else if (minutes > 0) {
    return `${minutes}m ${seconds % 60}s`;
  } else {
    return `${seconds}s`;
  }
}

/**
 * 格式化持续时间 (秒 → MM:SS 格式)
 */
export function formatDurationMMSS(seconds: number): string {
  const mins = Math.floor(seconds / 60);
  const secs = seconds % 60;
  return `${mins.toString().padStart(2, '0')}:${secs.toString().padStart(2, '0')}`;
}

/**
 * 获取事件显示图标 (返回 React 组件)
 */
export function getEventIcon(event: UnifiedEvent): React.ReactNode {
  // 优先使用事件类型图标
  const typeInfo = eventTypeConfig[event.type];
  if (typeInfo?.iconComponent) return typeInfo.iconComponent;

  // 其次使用来源图标
  const sourceInfo = sourceConfig[event.source];
  if (sourceInfo?.iconComponent) return sourceInfo.iconComponent;

  // 最后使用级别图标
  const levelInfo = levelConfig[event.level];
  if (levelInfo?.iconComponent) return levelInfo.iconComponent;

  return React.createElement('span', null, '•');
}

/**
 * 获取事件显示颜色
 */
export function getEventColor(event: UnifiedEvent): string {
  // 优先级: level error/fatal > category > source
  if (event.level === 'error' || event.level === 'fatal') {
    return levelConfig[event.level].color;
  }
  if (event.level === 'warn') {
    return levelConfig.warn.color;
  }
  return categoryConfig[event.category]?.color || sourceConfig[event.source]?.color || 'default';
}

/**
 * 判断事件是否为关键事件
 */
export function isCriticalEvent(event: UnifiedEvent): boolean {
  return (
    event.level === 'error' ||
    event.level === 'fatal' ||
    event.type === 'app_crash' ||
    event.type === 'app_anr'
  );
}

// Helper functions
function pad(n: number): string {
  return n.toString().padStart(2, '0');
}

function pad3(n: number): string {
  return n.toString().padStart(3, '0');
}

File: event_types.go
package main

import (
	"encoding/json"
)

// ========================================
// Event Source - 事件来源
// ========================================

type EventSource string

const (
	SourceLogcat    EventSource = "logcat"    // 设备日志
	SourceNetwork   EventSource = "network"   // 网络请求
	SourceDevice    EventSource = "device"    // 设备状态
	SourceApp       EventSource = "app"       // 应用生命周期
	SourceUI        EventSource = "ui"        // UI 状态变化
	SourceTouch     EventSource = "touch"     // 触摸事件
	SourceWorkflow  EventSource = "workflow"  // 自动化流程
	SourcePerf      EventSource = "perf"      // 性能指标
	SourceSystem    EventSource = "system"    // 系统事件
	SourceAssertion EventSource = "assertion" // 断言结果
	SourcePlugin    EventSource = "plugin"    // 插件生成的事件
)

// ========================================
// Event Category - 事件大类
// ========================================

type EventCategory string

const (
	CategoryLog         EventCategory = "log"
	CategoryNetwork     EventCategory = "network"
	CategoryState       EventCategory = "state"
	CategoryInteraction EventCategory = "interaction"
	CategoryAutomation  EventCategory = "automation"
	CategoryDiagnostic  EventCategory = "diagnostic"
	CategoryPlugin      EventCategory = "plugin" // 插件生成的事件
)

// ========================================
// Event Level - 事件级别
// ========================================

type EventLevel string

const (
	LevelVerbose EventLevel = "verbose"
	LevelDebug   EventLevel = "debug"
	LevelInfo    EventLevel = "info"
	LevelWarn    EventLevel = "warn"
	LevelError   EventLevel = "error"
	LevelFatal   EventLevel = "fatal"
)

// ========================================
// UnifiedEvent - 统一事件结构
// ========================================

type UnifiedEvent struct {
	// === 标识字段 ===
	ID        string `json:"id"`        // UUID
	SessionID string `json:"sessionId"` // 所属会话
	DeviceID  string `json:"deviceId"`  // 设备 ID

	// === 时间字段 ===
	Timestamp    int64 `json:"timestamp"`          // Unix 毫秒 (绝对时间)
	RelativeTime int64 `json:"relativeTime"`       // 相对 Session 开始的毫秒偏移
	Duration     int64 `json:"duration,omitempty"` // 持续时间 (ms)

	// === 分类字段 ===
	Source   EventSource   `json:"source"`   // 事件来源
	Category EventCategory `json:"category"` // 事件大类
	Type     string        `json:"type"`     // 具体类型 (如 "http_request", "activity_start")
	Level    EventLevel    `json:"level"`    // 事件级别

	// === 内容字段 ===
	Title   string `json:"title"`             // 简短标题 (用于列表显示)
	Summary string `json:"summary,omitempty"` // 摘要 (可搜索)

	// === 扩展数据 (JSON) ===
	Data   json.RawMessage `json:"data,omitempty"`   // 类型特定的详细数据
	Detail json.RawMessage `json:"detail,omitempty"` // 别名，向后兼容

	// === 关联字段 ===
	ParentID string `json:"parentId,omitempty"` // 父事件 ID
	StepID   string `json:"stepId,omitempty"`   // Workflow Step ID
	TraceID  string `json:"traceId,omitempty"`  // 跨事件追踪 ID

	// === 聚合信息 ===
	AggregateCount int   `json:"aggregateCount,omitempty"` // 聚合的事件数量
	AggregateFirst int64 `json:"aggregateFirst,omitempty"` // 聚合的第一条时间
	AggregateLast  int64 `json:"aggregateLast,omitempty"`  // 聚合的最后一条时间

	// === 插件扩展字段 ===
	Tags              []string               `json:"tags,omitempty"`              // 标签（插件可添加）
	Metadata          map[string]interface{} `json:"metadata,omitempty"`          // 元数据（插件可附加）
	ParentEventID     string                 `json:"parentEventId,omitempty"`     // 派生事件的父事件 ID
	GeneratedByPlugin string                 `json:"generatedByPlugin,omitempty"` // 生成此事件的插件 ID
}

// ========================================
// DeviceSession - 设备会话
// ========================================

// SessionConfig - Session 启动配置
type SessionConfig struct {
	Logcat    LogcatConfig    `json:"logcat"`
	Recording RecordingConfig `json:"recording"`
	Proxy     ProxyConfig     `json:"proxy"`
	Monitor   MonitorConfig   `json:"monitor"`
}

type LogcatConfig struct {
	Enabled       bool   `json:"enabled"`
	PackageName   string `json:"packageName,omitempty"`
	PreFilter     string `json:"preFilter,omitempty"`
	ExcludeFilter string `json:"excludeFilter,omitempty"`
}

type RecordingConfig struct {
	Enabled bool   `json:"enabled"`
	Quality string `json:"quality,omitempty"` // "low", "medium", "high"
}

type ProxyConfig struct {
	Enabled     bool `json:"enabled"`
	Port        int  `json:"port,omitempty"`
	MitmEnabled bool `json:"mitmEnabled,omitempty"`
}

type MonitorConfig struct {
	Enabled bool `json:"enabled"` // 设备状态监控 (电池、网络、屏幕、应用生命周期)
}

type DeviceSession struct {
	ID         string `json:"id"`
	DeviceID   string `json:"deviceId"`
	Type       string `json:"type"` // "manual", "workflow", "recording", "debug", "auto"
	Name       string `json:"name"`
	StartTime  int64  `json:"startTime"` // Unix ms
	EndTime    int64  `json:"endTime"`   // 0 = active
	Status     string `json:"status"`    // "active", "completed", "failed", "cancelled"
	EventCount int    `json:"eventCount"`

	// Session config (启用的功能)
	Config SessionConfig `json:"config"`

	// Recording info
	VideoPath     string `json:"videoPath,omitempty"`
	VideoDuration int64  `json:"videoDuration,omitempty"`
	VideoOffset   int64  `json:"videoOffset,omitempty"` // 视频相对 session 开始的偏移

	// Metadata
	Metadata map[string]any `json:"metadata,omitempty"`
}

// ========================================
// Type-Specific Data Structures
// ========================================

// LogcatData logcat 事件数据
type LogcatData struct {
	Tag          string `json:"tag"`
	Message      string `json:"message"`
	AndroidLevel string `json:"androidLevel"` // V/D/I/W/E/F
	PID          int    `json:"pid,omitempty"`
	TID          int    `json:"tid,omitempty"`
	PackageName  string `json:"packageName,omitempty"`
	Raw          string `json:"raw,omitempty"`
}

// ========================================
// Time Index Entry - 时间索引条目
// ========================================

type TimeIndexEntry struct {
	Second       int    `json:"second"`       // 相对 session 开始的秒数
	EventCount   int    `json:"eventCount"`   // 该秒内的事件数量
	FirstEventID string `json:"firstEventId"` // 该秒内第一个事件的 ID
	HasError     bool   `json:"hasError"`     // 该秒内是否有错误事件
}

// ========================================
// Bookmark - 时间书签
// ========================================

type Bookmark struct {
	ID           string `json:"id"`
	SessionID    string `json:"sessionId"`
	RelativeTime int64  `json:"relativeTime"` // 相对 session 开始的毫秒
	Label        string `json:"label"`
	Color        string `json:"color,omitempty"`
	Type         string `json:"type"` // user, error, milestone, assertion_fail
	CreatedAt    int64  `json:"createdAt"`
}

// ========================================
// Event Type Registry - 事件类型注册表
// ========================================

type EventTypeInfo struct {
	Type        string        // 类型标识
	Source      EventSource   // 来源
	Category    EventCategory // 大类
	Description string        // 描述
}

var EventRegistry = map[string]EventTypeInfo{
	// === Logcat 事件 ===
	"logcat": {
		Type: "logcat", Source: SourceLogcat, Category: CategoryLog,
		Description: "Android logcat output",
	},
	"logcat_aggregated": {
		Type: "logcat_aggregated", Source: SourceLogcat, Category: CategoryLog,
		Description: "Aggregated logcat entries",
	},

	// === Network 事件 ===
	"http_request": {
		Type: "http_request", Source: SourceNetwork, Category: CategoryNetwork,
		Description: "DEPRECATED: Use 'network_request' instead. HTTP/HTTPS request (legacy name)",
	},
	"network_request": {
		Type: "network_request", Source: SourceNetwork, Category: CategoryNetwork,
		Description: "Network request (proxy-captured) - Standard event type for all HTTP/HTTPS traffic",
	},
	"websocket_message": {
		Type: "websocket_message", Source: SourceNetwork, Category: CategoryNetwork,
		Description: "WebSocket message",
	},

	// === Device State 事件 ===
	"battery_change": {
		Type: "battery_change", Source: SourceDevice, Category: CategoryState,
		Description: "Battery level or status change",
	},
	"network_change": {
		Type: "network_change", Source: SourceDevice, Category: CategoryState,
		Description: "Network connectivity change",
	},
	"screen_change": {
		Type: "screen_change", Source: SourceDevice, Category: CategoryState,
		Description: "Screen state change",
	},

	// === App Lifecycle 事件 ===
	"app_start": {
		Type: "app_start", Source: SourceApp, Category: CategoryState,
		Description: "Application started",
	},
	"app_stop": {
		Type: "app_stop", Source: SourceApp, Category: CategoryState,
		Description: "Application stopped",
	},
	"activity_start": {
		Type: "activity_start", Source: SourceApp, Category: CategoryState,
		Description: "Activity started",
	},
	"activity_stop": {
		Type: "activity_stop", Source: SourceApp, Category: CategoryState,
		Description: "Activity stopped",
	},
	"app_crash": {
		Type: "app_crash", Source: SourceApp, Category: CategoryState,
		Description: "Application crash",
	},
	"app_anr": {
		Type: "app_anr", Source: SourceApp, Category: CategoryState,
		Description: "Application Not Responding",
	},

	// === Touch 事件 ===
	"touch": {
		Type: "touch", Source: SourceTouch, Category: CategoryInteraction,
		Description: "Touch event",
	},
	"gesture": {
		Type: "gesture", Source: SourceTouch, Category: CategoryInteraction,
		Description: "Recognized gesture",
	},

	// === Workflow 事件 ===
	"workflow_start": {
		Type: "workflow_start", Source: SourceWorkflow, Category: CategoryAutomation,
		Description: "Workflow execution started",
	},
	"workflow_step_start": {
		Type: "workflow_step_start", Source: SourceWorkflow, Category: CategoryAutomation,
		Description: "Workflow step started",
	},
	"workflow_step_end": {
		Type: "workflow_step_end", Source: SourceWorkflow, Category: CategoryAutomation,
		Description: "Workflow step completed",
	},
	"workflow_complete": {
		Type: "workflow_complete", Source: SourceWorkflow, Category: CategoryAutomation,
		Description: "Workflow execution completed",
	},
	"workflow_error": {
		Type: "workflow_error", Source: SourceWorkflow, Category: CategoryAutomation,
		Description: "Workflow execution error",
	},

	// === Performance 事件 ===
	"perf_sample": {
		Type: "perf_sample", Source: SourcePerf, Category: CategoryDiagnostic,
		Description: "Performance metric sample",
	},

	// === Assertion 事件 ===
	"assertion_result": {
		Type: "assertion_result", Source: SourceAssertion, Category: CategoryAutomation,
		Description: "Assertion evaluation result",
	},

	// === Breakpoint 事件 ===
	"breakpoint_hit": {
		Type: "breakpoint_hit", Source: SourceNetwork, Category: CategoryNetwork,
		Description: "Request/response paused at breakpoint",
	},
	"breakpoint_resolved": {
		Type: "breakpoint_resolved", Source: SourceNetwork, Category: CategoryNetwork,
		Description: "Breakpoint resolved (forwarded/dropped/modified)",
	},

	// === System 事件 ===
	"session_start": {
		Type: "session_start", Source: SourceSystem, Category: CategoryState,
		Description: "Session started",
	},
	"session_end": {
		Type: "session_end", Source: SourceSystem, Category: CategoryState,
		Description: "Session ended",
	},
	"recording_start": {
		Type: "recording_start", Source: SourceSystem, Category: CategoryState,
		Description: "Screen recording started",
	},
	"recording_end": {
		Type: "recording_end", Source: SourceSystem, Category: CategoryState,
		Description: "Screen recording ended",
	},
}

// ParseEventLevel converts a string level (e.g. from old SessionEvent) to EventLevel.
// Returns LevelInfo for unrecognized values.
func ParseEventLevel(level string) EventLevel {
	switch level {
	case "fatal":
		return LevelFatal
	case "error":
		return LevelError
	case "warn":
		return LevelWarn
	case "info":
		return LevelInfo
	case "debug":
		return LevelDebug
	case "verbose":
		return LevelVerbose
	default:
		return LevelInfo
	}
}

// GetCategoryForType 根据事件类型获取分类
func GetCategoryForType(eventType string) EventCategory {
	if info, ok := EventRegistry[eventType]; ok {
		return info.Category
	}
	return CategoryDiagnostic
}
